presto-spi/src/main/java/com/facebook/presto/spi/statistics/ColumnStatistics.java

```diff
@@ -11,56 +11,52 @@
 
 import static com.facebook.presto.spi.statistics.Estimate.unknownValue;
-import static java.util.Collections.unmodifiableMap;
+import static java.util.Collections.singletonList;
+import static java.util.Collections.unmodifiableList;
 import static java.util.Objects.requireNonNull;
 
 public final class ColumnStatistics
 {
-    private final Map<String, Estimate> statistics;
-    private static final String DATA_SIZE_STATISTIC_KEY = "data_size";
-    private static final String NULLS_FRACTION_STATISTIC_KEY = "nulls_fraction";
-    private static final String DISTINCT_VALUES_STATITIC_KEY = "distinct_values_count";
+    private static final List<RangeColumnStatistics> SINGLE_UNKNOWN_RANGE_STATISTICS = singletonList(RangeColumnStatistics.builder().build());
 
-    private ColumnStatistics(Estimate dataSize, Estimate nullsCount, Estimate distinctValuesCount)
-    {
-        requireNonNull(dataSize, "dataSize can not be null");
-        statistics = createStatisticsMap(dataSize, nullsCount, distinctValuesCount);
-    }
+    private final List<RangeColumnStatistics> rangeColumnStatistics;
+    private final Estimate nullsFraction;
 
-    private static Map<String, Estimate> createStatisticsMap(Estimate dataSize, Estimate nullsFraction, Estimate distinctValuesCount)
+    private ColumnStatistics(Estimate nullsFraction, List<RangeColumnStatistics> rangeColumnStatistics)
     {
-        Map<String, Estimate> statistics = new HashMap<>();
-        statistics.put(DATA_SIZE_STATISTIC_KEY, dataSize);
-        statistics.put(NULLS_FRACTION_STATISTIC_KEY, nullsFraction);
-        statistics.put(DISTINCT_VALUES_STATITIC_KEY, distinctValuesCount);
-        return unmodifiableMap(statistics);
-    }
+        this.nullsFraction = requireNonNull(nullsFraction, "nullsFraction can not be null");
+        requireNonNull(rangeColumnStatistics, "rangeColumnStatistics can not be null");
+        if (rangeColumnStatistics.size() > 1) {
+            // todo add support for multiple ranges.
+            throw new IllegalArgumentException("Statistics for multiple ranges are not supported");
+        }
+        if (rangeColumnStatistics.isEmpty()) {
+            rangeColumnStatistics = SINGLE_UNKNOWN_RANGE_STATISTICS;
+        }
 
-    public Estimate getDataSize()
-    {
-        return statistics.get(DATA_SIZE_STATISTIC_KEY);
-    }
+        if (nullsFraction.isValueUnknown() ^ rangeColumnStatistics.get(0).getFraction().isValueUnknown()) {
+            throw new IllegalArgumentException("All or none fraction/nullsFraction must be set");
+        }
 
-    public Estimate getNullsFraction()
-    {
-        return statistics.get(NULLS_FRACTION_STATISTIC_KEY);
+        this.rangeColumnStatistics = unmodifiableList(rangeColumnStatistics);
     }
 
-    public Estimate getDistinctValuesCount()
+    public RangeColumnStatistics getOnlyRangeColumnStatistics()
     {
-        return statistics.get(DISTINCT_VALUES_STATITIC_KEY);
+        return rangeColumnStatistics.get(0);
```

Comment:
i would put (repeat) the `size()==1` check here too, but probably it's against convention